packages/payment-processor/src/payment/erc20.ts

```diff
@@ -18,34 +18,45 @@ import {
   getSigner,
   validateRequest,
 } from './utils';
+import { ICurrency } from '@requestnetwork/types/dist/request-logic-types';
+import { ISwapSettings } from '..';
 
 /**
  * Processes a transaction to pay an ERC20 Request.
  * @param request
  * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
  * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
  * @param feeAmount optionally, the fee amount to pay. Only applicable to ERC20 Fee Payment network. Defaults to the fee amount.
+ * @param swapSettings optionally, the settings to swap a maximum amount of currency, through a swap path, before a deadline, to pay
  * @param overrides optionally, override default transaction values, like gas.
  */
 export async function payErc20Request(
```

Comment:
This function signature is starting to look too big. All the `undefined` we have to declare `payRequest` shows an issue.
I think we need something like an `options` object parameter.
The CodeClimate check is not mandatory, but has good hints about this kind of things.